enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,325 @@
+---
+title: automated-backups-no-config
+authors:
+  - "@elbehery"
+reviewers:
+  - "@JoelSpeed"
+  - "@dusk125"
+  - "@hasbro17"
+  - "@tjungblu"
+  - "@vrutkovs"
+approvers:
+  - "@hasbro17"
+api-approvers:
+  - "@JoelSpeed"
+creation-date: 2024-06-17
+last-updated: 2024-10-21
+tracking-link:
+  - https://issues.redhat.com/browse/ETCD-609
+see-also:
+  - "https://issues.redhat.com/browse/OCPSTRAT-1411"
+  - "https://issues.redhat.com/browse/OCPSTRAT-529"
+---
+
+
+# Automated Backups of etcd with No Config
+
+## Summary
+
+To enable automated backups of Etcd database and cluster resources of an Openshift cluster from day one after installation, 
+without provided configuration from the user.
+
+This document outlines the possible approaches that were investigated, discussed and tested, in addition to the final implemented approach.  
+
+## Motivation
+
+The current method of **Backup** and **Restore** of an Openshift cluster is outlined in  [Automated Backups of etcd](https://github.com/openshift/enhancements/blob/master/enhancements/etcd/automated-backups.md). 
+This approach relies on user supplied configuration, in the form of `config.openshift.io/v1alpha1 Backup` CR. This method utilizes `batchv1.CronJob` in order to take Backups of Etcd and cluster resources.
+
+To improve the user's experience, this work proposes taking etcd backup using default configurations from day one, without user interference.
+
+Moreover, for recovery in disaster scenarios, where more than one control-plane node is lost from an Openshift cluster, this proposal attempts to store backups on all control-plane nodes.
+Having backups of an Openshift cluster on all control-plane nodes, improves the recovery probability in disaster scenarios.
+
+### User Stories
+
+* As a cluster administrator, I would like an OCP cluster backups to be taken without user's supplied configuration from day one, after installation.
+* As a cluster administrator, I would like to an OCP cluster backups available on all master nodes, so that I have a recent cluster state to recover from in the event of losing control-plane nodes.
+
+### Goals
+
+* Backups should be taken without configuration after cluster installation from day 1.
+* Backups are saved to a default location on each master node's file system.
+* Backups are taken according to a default schedule.
+* Backups are being maintained according to a default retention policy.
+
+### Non-Goals
+
+* Save cluster backups to remote cloud storage (e.g. S3 Bucket).
+  - This could be a future enhancement or extension to the API.
+  - Having backups in an independent storage from OCP cluster is crucial for Disaster Recovery scenarios.
+* Automate cluster restoration.
+* Provide automated backups for non-self hosted architectures like Hypershift.
```

Comment:
I'm late to the party here, but it's probably worth some text on this exclusion down in the:

> ```markdown
> #### Hypershift / Hosted Control Planes
> This proposal does not affect HyperShift.
> ```

section.  Because while it's possible that you can implement for standalone, and then do something similar for HyperShift (or MicroShift), those folks will still _want_ automated backups eventually, right?  And it's good to at least have a rough idea of how automated backups would work on those systems.  Would you use a completely different type?  Would you stick with `Backup.config.openshift.io`, and drop your current `+genclient:nonNamespaced` and have `pvcName` be required on HyperShift?  Some kind of plan.

Reply:
great point, wdyt  @sjenning 🤔

Reply:
Yes, hypershift does not currently run the etcd-operator so this would not apply to it

Reply:
We do want to eventually have automatic backups in HyperShift, but agree that we would want a different API possibly within the HostedCluster CR.

One thing that I would like to see called out here though is that because the Backup CRD is not actually useful in a HyperShift hosted cluster, it should not be applied by the CVO to clusters with an external control plane.